src/topologies/halfedge.jl

```diff
@@ -220,49 +216,50 @@ function HalfEdgeTopology(elems::AbstractVector{<:Connectivity}; sort=true)
 end
 
 function adjsortperm(elems::AbstractVector{<:Connectivity})
-  listi = collect(eachindex(elems)[2:end])
   # initialize list of adjacent elements
   # with first element from original list
-  adjs = Int[1]
-  sizehint!(adjs, length(elems))
+  einds = Int[1]
+  sizehint!(einds, length(elems))
+
+  # remaining list of elements to process
+  oinds = collect(2:length(elems))
```

Comment:
This is less safe than `collect(eachindex(elems)[2:end])`, because the `elems::AbstractVector` type parameter allows a vector with a `firstindex !== 1`.

```suggestion
oinds = collect(eachindex(elems)[2:end])
```

Reply:
Isn't the algorithm written in terms of a local enumeration already? We initialize `einds = Int[1]` to mean the first element in the vector of elements, and then enumerate the remaining elements as `2:N`? Later on, we index `oinds[iter]` where `iter` starts as `1` too.

Reply:
Ah good catch, properly `einds` should be initialized as `einds = Int[firstindex(elems)]` along with the above. All the other indexing is into `oinds` which is a regular `Vector` (i.e. starts at `1`) or indexing into `elems` using (ultimately) values from `oinds` which, with the above, is filled with guaranteed inbounds values produced by `eachindex`.

Reply:
Good point. Feel free to adjust 👍🏽